packages/core/src/amazonqGumby/commands.ts

```diff
@@ -5,28 +5,41 @@
 
 import * as vscode from 'vscode'
 import { telemetry } from '../shared/telemetry/telemetry'
-import { ExtContext } from '../shared/extensions'
 import { Commands } from '../shared/vscode/commands2'
 import * as CodeWhispererConstants from '../codewhisperer/models/constants'
-import { startTransformByQWithProgress, confirmStopTransformByQ } from '../codewhisperer/commands/startTransformByQ'
+import { stopTransformByQ } from '../codewhisperer/commands/startTransformByQ'
 import { transformByQState } from '../codewhisperer/models/model'
 import { AuthUtil } from '../codewhisperer/util/authUtil'
 import { CancelActionPositions, logCodeTransformInitiatedMetric } from './telemetry/codeTransformTelemetry'
+import { ChatControllerEventEmitters } from './chat/controller/controller'
+import { focusAmazonQPanel } from '../auth/ui/vue/show'
+import { sleep } from '../shared/utilities/timeoutUtils'
+import { randomUUID } from 'crypto'
 
 export const showTransformByQ = Commands.declare(
     { id: 'aws.awsq.transform', compositeKey: { 0: 'source' } },
-    (context: ExtContext) => async (source: string) => {
+    (controllerEventEmitters: ChatControllerEventEmitters) => async (source: string) => {
         if (AuthUtil.instance.isConnectionExpired()) {
             await AuthUtil.instance.notifyReauthenticate()
         }
 
         if (transformByQState.isNotStarted()) {
             logCodeTransformInitiatedMetric(source)
-            await startTransformByQWithProgress()
+            await focusAmazonQPanel().then(async () => {
+                // non blocking wait for a tenth of a second
+                // note: we need this because the webview has to be loaded
+                // before it can listen to events we fire
+                await sleep(100)
```

Comment:
Maybe increase to 250 or something to be safe

Reply:
Note: Looking at Nielsons Heuristics linked above, a "large move" or "complex easing" can be 200-500ms.

I would generalize a panel display is a "large" ux move.